CMakeLists.txt

```diff
@@ -1175,10 +1178,14 @@ status("    Java tests:"    BUILD_TESTS AND opencv_test_java_BINARY_DIR
 
 # ========================= matlab =========================
 status("")
-status("  Matlab:")
-status("    mex:"         MATLAB_MEX_SCRIPT  THEN  "${MATLAB_MEX_SCRIPT}"   ELSE NO)
-if (MATLAB_FOUND)
-  status("    Compiler/generator:" MEX_WORKS    THEN  "Working"                ELSE "Not working (bindings will not be generated)")
+if(WITH_MATLAB AND MATLAB_FOUND)
```

Comment:
Please use here `if(MATLAB_FOUND)` only. `WITH_MATLAB` is unnecessary here (and sometimes buggy)

Also remove nested `if(MATLAB_FOUND)` check and the "else" branch below (or replace it to `elseif(WITH_MATLAB)`).